source/s3_request_messages.c

```diff
@@ -266,7 +266,7 @@ struct aws_http_message *aws_s3_create_multipart_upload_message_new(
         false /*exclude_x_amz_meta*/);
 
     if (message == NULL) {
-        goto error_clean_up;
```

Comment:
That's the first thing in the function, everything in the function is based on the message. 

As long as people in the future be careful about error handling, we should be good. Just make sure before any allocation, or the first allocation failed, just return, otherwise, clean up.